src/commands/webSocketTerminal.ts

```diff
@@ -0,0 +1,660 @@
+          submit = true;
+          char = char.slice(0, -1);
+        }
+        // Replace all single \r with \r\n (prompt) or space (read)
+        char = char.replace(/\r/g, this._state == "prompt" ? "\r\n" : " ");
+        const inputArr = this._input.split("\r\n");
+        if (this._cursorCol < this._margin + inputArr[inputArr.length - 1].length) {
+          // Insert the new char(s)
+          inputArr[inputArr.length - 1] = `${inputArr[inputArr.length - 1].slice(
+            0,
+            this._cursorCol - this._margin
+          )}${char}${inputArr[inputArr.length - 1].slice(this._cursorCol - this._margin)}`;
+          this._input = inputArr.join("\r\n");
+          this._cursorCol += char.length;
+          this._hideCursorWrite(`\x1b[4h${char.replace(/\r\n/g, `\r\n${this._multiLinePrompt}`)}\x1b[4l`);
+        } else {
+          // Append the new char(s)
+          this._input += char;
+          this._cursorCol += char.length;
+          this._hideCursorWrite(char.replace(/\r\n/g, `\r\n${this._multiLinePrompt}`));
+        }
+        if (submit) {
+          if (this._state == "prompt") {
+            // Reset historyIdx
+            this._historyIdx = -1;
+
+            if (this._input != "" && !this._input.includes("\r\n")) {
+              // Remove the input from the existing history
+              this._history = this._history.filter((h) => h != this._input);
+
+              // Append this input to the history
+              this._history.push(this._input);
+            }
+
+            // Reset first line tracker
+            this._firstOutputLineSincePrompt = true;
+          } else {
+            // Reset first line tracker
+            this._firstOutputLineSincePrompt = false;
+          }
+
+          // Send the input to the server for processing
+          this._socket.send(JSON.stringify({ type: this._state, input: this._input }));
+          if (this._state == "prompt") {
+            this._hideCursorWrite(`\x1b]633;C\x07\x1b]633;E;${this._inputEscaped()}\x07\r\n`);
+            if (this._input == "") {
+              this._promptExitCode = "";
+            }
+          }
+          this._input = "";
+          this._state = "eval";
+        } else if (this._input != "" && this._state == "prompt" && this._syntaxColoringEnabled()) {
+          // Syntax color input
+          this._socket.send(JSON.stringify({ type: "color", input: this._input }));
+        }
+      }
+    }
+  }
+}
+
+function terminalConfigForUri(api: AtelierAPI, extensionUri: vscode.Uri): vscode.ExtensionTerminalOptions | undefined {
+  // Make sure the server connection is active
+  if (!api.active || api.ns == "") {
+    vscode.window.showErrorMessage("WebSocket Terminal requires an active server connection.", "Dismiss");
+    return;
+  }
+  // Make sure the server has the terminal endpoint
+  if (api.config.apiVersion < 7) {
+    vscode.window.showErrorMessage("WebSocket Terminal requires Atelier API version 7 or above.", "Dismiss");
```

Comment:
I suggest stating the IRIS version rather than the API version, as the latter is meaningless to most.